File: packages/ios/ApiContracts/Sources/ApiContracts/Models/UploadModels.swift
import Foundation

// MARK: - Upload Response

/// POST /api/upload 响应体（服务端直传兼容）
public struct UploadResponse: Codable, Sendable {
    public let url: String
    public let key: String

    public init(url: String, key: String) {
        self.url = url
        self.key = key
    }
}

/// POST /api/upload/presign 请求体
public struct PresignUploadRequest: Codable, Sendable {
    public let contentType: String

    public init(contentType: String) {
        self.contentType = contentType
    }
}

/// POST /api/upload/presign 响应体
public struct PresignUploadResponse: Codable, Sendable {
    public let uploadUrl: String
    public let key: String
    public let url: String
}

File: packages/android/apicontracts/src/main/kotlin/com/aistudy/apicontracts/UploadModels.kt
package com.aistudy.apicontracts

import kotlinx.serialization.Serializable

// MARK: - Upload
// POST /api/upload/presign → 预签名 URL，PUT 到对象存储后再回调
// POST /api/upload       → 直传（base64 / 小文件）

@Serializable
data class UploadResponse(
    val url: String,
    val key: String,
)

@Serializable
data class PresignUploadRequest(
    val contentType: String,
)

@Serializable
data class PresignUploadResponse(
    val uploadUrl: String,
    val key: String,
    val url: String,
)
